fedimint-recurringd/src/db.rs

```diff
@@ -182,3 +201,131 @@ impl_db_lookup!(
     key = PaymentCodeInvoiceKey,
     query_prefix = PaymentCodeInvoicePrefix
 );
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Encodable, Decodable)]
+pub struct SchemaVersionKey;
+
+impl_db_record!(
+    key = SchemaVersionKey,
+    value = u64,
+    db_prefix = DbKeyPrefix::SchemaVersion,
+);
+
+type DbMigration =
+    for<'a> fn(&'a RecurringInvoiceServer, DatabaseTransaction<'a>) -> BoxFuture<'a, ()>;
+
+impl RecurringInvoiceServer {
+    pub(crate) fn migrations() -> BTreeMap<u64, DbMigration> {
+        vec![(
+            1,
+            (|server: &RecurringInvoiceServer, dbtx| Box::pin(server.db_migration_v1(dbtx)))
+                as DbMigration,
+        )]
+        .into_iter()
+        .collect()
+    }
+
+    /// Backfill DB fix for bug that caused "holes" in invoice indices keeping
+    /// the client from syncing. See <https://github.com/fedimint/fedimint/pull/7653>.
+    async fn db_migration_v1(&self, mut dbtx: DatabaseTransaction<'_>) {
+        const BACKFILL_AMOUNT: Amount = Amount::from_msats(111111);
+
+        let mut payment_codes = dbtx
+            .find_by_prefix(&PaymentCodePrefix)
+            .await
+            .map(|(k, v)| (k.payment_code_id, v))
+            .collect::<HashMap<PaymentCodeId, PaymentCodeEntry>>()
```

Comment:
Style unimportant comment: IMO explicit type on `let xyz : Type =` is better than on `.collect::<Type>()`. Just looks cleaner and is in the more expected place. Not sure if `.await` here prevents it or something.

Reply:
Good to know, I just always did it the other way around out of habit :see_no_evil: Fixed.